composer/lib/dependabot/composer/helpers.rb

```diff
@@ -54,22 +74,26 @@ def self.composer_version(composer_json, parsed_lockfile = nil)
         V2
       end
 
+      sig { params(message: String).returns(T.nilable(String)) }
       def self.dependency_url_from_git_clone_error(message)
-        if message.match?(FAILED_GIT_CLONE_WITH_MIRROR)
-          dependency_url = message.match(FAILED_GIT_CLONE_WITH_MIRROR).named_captures.fetch("url")
-          raise "Could not parse dependency_url from git clone error: #{message}" if dependency_url.empty?
+        extract_and_clean_dependency_url(message, FAILED_GIT_CLONE_WITH_MIRROR) ||
+          extract_and_clean_dependency_url(message, FAILED_GIT_CLONE)
+      end
 
-          clean_dependency_url(dependency_url)
-        elsif message.match?(FAILED_GIT_CLONE)
-          dependency_url = message.match(FAILED_GIT_CLONE).named_captures.fetch("url")
-          raise "Could not parse dependency_url from git clone error: #{message}" if dependency_url.empty?
```

Comment:
Since `if message.match?(FAILED_GIT_CLONE_WITH_MIRROR)` condition and `elsif message.match?(FAILED_GIT_CLONE)` were same except being different error message so the check is moved to a function `extract_and_clean_dependency_url` to remove duplications.